resources/packs/aws/aws_ec2.go

```diff
@@ -735,30 +717,52 @@ func (s *mqlAwsEc2) gatherInstanceInfo(instances []types.Reservation, imdsvVersi
 				"httpTokens", httpTokens,
 				"state", string(instance.State.Name),
 				"deviceMappings", mqlDevices,
-				"securityGroups", sgs,
 				"publicDnsName", core.ToString(instance.PublicDnsName),
 				"stateReason", stateReason,
 				"stateTransitionReason", core.ToString(instance.StateTransitionReason),
 				"ebsOptimized", core.ToBool(instance.EbsOptimized),
 				"instanceType", string(instance.InstanceType),
 				"tags", Ec2TagsToMap(instance.Tags),
-				"image", mqlImage,
 				"launchTime", instance.LaunchTime,
 				"privateIp", core.ToString(instance.PrivateIpAddress),
 				"privateDnsName", core.ToString(instance.PrivateDnsName),
 			}
+			sgs := []interface{}{}
+
+			for i := range instance.SecurityGroups {
+				// NOTE: this will create the resource and determine the data in its init method
+				mqlSg, err := s.MotorRuntime.CreateResource("aws.ec2.securitygroup",
+					"arn", fmt.Sprintf(securityGroupArnPattern, regionVal, account.ID, core.ToString(instance.SecurityGroups[i].GroupId)),
+				)
+				if err == nil {
+					sgs = append(sgs, mqlSg)
+				} else {
+					log.Warn().Err(err).Msg("unable to create security groups resource for instance")
```

Comment:
This does not feel correct to me. 

I think we need to change:

```
// List of security groups (ids) associated with the instance
securityGroups []aws.ec2.securitygroup
```

to lazy load:

```
// List of security groups (ids) associated with the instance
securityGroups() []aws.ec2.securitygroup
```

This would still show the error but users can see the rest of ec2 instance. I suggest we store the security groups in the cache of the resources and just use that one when MQL calls the GetSecurityGroups() methods